syncmap_test.go

```diff
@@ -0,0 +1,212 @@
+package dapper_test
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+	"strings"
+	"sync"
+	"testing"
+
+	. "github.com/dogmatiq/dapper"
+)
+
+type syncmaps struct {
+	Map sync.Map
+}
+
+// This test verifies that that sync.Map key/value types are always rendered.
+func TestPrinter_SyncMap(t *testing.T) {
+	var m sync.Map
+
+	test(t, "empty sync.Map", &m, "*sync.Map{}")
+
+	m.Store(1, 100)
+	m.Store(2, 200)
+
+	test(
+		t,
+		"sync.Map",
+		&m,
+		"*sync.Map{",
+		"    int(1): int(100)",
+		"    int(2): int(200)",
+		"}",
+	)
+}
+
+// This test verifies the formatting of sync.Map key/values in the named structs.
+func TestPrinter_SyncMapInNamedStruct(t *testing.T) {
+	test(
+		t,
+		"empty sync.Map",
+		&syncmaps{},
+		"*github.com/dogmatiq/dapper_test.syncmaps{<zero>}",
```

Comment:
This is not testing the rendering of any empty sync map... You'll need to add a `Force bool` field to the `syncmaps` type like I did in the `maps` one to avoid the collapsing to `<zero>`.

See:

- https://github.com/dogmatiq/dapper/blob/master/map_test.go#L13
- https://github.com/dogmatiq/dapper/blob/master/map_test.go#L41